### DISCUSSION TITLE
CS131,Homewrok0,Basics | SHEN's BLOG

### DISCUSSION THREAD

**Discussion by veraposeidon** [veraposeidon/veraposeidon.github.io#94] [General]
https://shenxiaohai.me/2018/04/15/cs131-homework0/

**Comment by xuezha007** [comment]
我现在也在学cs131  哥们 加下我qq 251525317 咱俩有不懂的互相交流一下

**Comment by veraposeidon** [comment]
已加

**Comment by LoveGalaxy** [comment]
兄弟，无意冒犯，但是感觉错误有点多啊，M的定义都错了，后面的结果就跟着有问题了，还有q1.3那个乘法是什么算法啊，我表示我一脸懵逼的在做题，a是（1,3）的b是（3,1）的向量，aT * b?什么操作这是

**Comment by veraposeidon** [comment]
@guruL 感谢指正，M的定义已经纠正，后面应该没有问题。
至于你说的Q1.3小题，题目只是让你熟悉一下python和numpy的操作，谈不上什么算法。
至于你不能理解的a和b相乘，推荐看一下
[matmul](https://docs.scipy.org/doc/numpy/reference/generated/numpy.matmul.html)的介绍，了解一维向量是如何与二维矩阵进行计算的。

**Comment by LoveGalaxy** [comment]
好的，感谢。关于后面几个颜色分量分解的数据，我看到函数里有这样的注释
out: numpy array of shape(image_height, image_width, 3)
然后如果把图片分解成rgb单一分量，又需要又3个通道的图片，是考虑讲别的通道设置为0，还是说，将三个通道都设为相同的值呢，像请教你怎么理解的。

**Comment by LoveGalaxy** [comment]
恩，上一个问题我看懂了，应该是排除某一个通道的。最后一个合成图应该是把两个图片和成在一起把，不是应该是拼接在一起

**Comment by veraposeidon** [comment]
@guruL 最后一题是两张图各拼一半，我审题没仔细，已经纠正。
